nilearn/datasets/atlas.py

```diff
@@ -377,31 +487,46 @@ def fetch_atlas_harvard_oxford(atlas_name, data_dir=None,
         names[int(label.get('index')) + 1] = label.text
     names = list(names.values())
 
-    if not symmetric_split:
-        return Bunch(maps=atlas_img, labels=names)
-
-    if atlas_name in ("cort-prob-1mm", "cort-prob-2mm",
-                      "sub-prob-1mm", "sub-prob-2mm"):
-        raise ValueError("Region splitting not supported for probabilistic "
-                         "atlases")
+    atlas_niimg = check_niimg(atlas_img)
 
-    atlas_img = check_niimg(atlas_img)
     if lateralized:
-        return Bunch(maps=atlas_img, labels=names)
+        return Bunch(maps=atlas_niimg, labels=names)
 
-    atlas = get_data(atlas_img)
+    if not symmetric_split:
+        atlas = get_data(atlas_niimg)
```

Comment:
this code block which is intended for renaming some labels in discrete atlases gets executed for the probabilistic atlases (both for harvard-oxford and juelich) as well. as a result

- it takes a very long time
- results are incorrect for Juelich (all values are roughly divided by 2) . they are correct for harvard-oxford by coincidence, because none of the labels need to be merged in that case.

Once this is fixed, could you check that the function runs  and returns the correct output for all the atlases it can fetch?


IMO this function is too long and convoluted, with a control flow that is hard to follow, which is why it is difficult to get it to work. I believe simplifying it or breaking it down to smaller functions may make things easier, both for writing it now and for reading and maintaining it later. a few short comments can be useful as well. using more explicit variable names eg `is_already_lateralized` instead of `lateralized` can also help